js/webapp.js

```diff
@@ -433,6 +429,45 @@ ImportManager.prototype = {
   
 }
 
+function getLoopbackDefaults(webappOptions) {
+  //if this isn't here, throw
+  let nodeConfig = webappOptions.serverConfig.node;
+
+  let getPreferredAddress = function(addresses) {
+    if (!addresses) {
+      return '127.0.0.1';
+    }
+    for (let i = 0; i < addresses.length; i++) {
+      let address = addresses[i];
+      if (address == '127.0.0.1' || address == 'localhost' || address == '0.0.0.0') {
```

Comment:
In this PR https://github.com/zowe/zss-auth/pull/8/files#diff-9c4028ca3fee8475c7b0f0a4f5a92580R115 I used 'ipaddr.js' to check whether an address is loopback: entire ranges of loopback addresses  are defined, so, e.g. 127.0.0.2 is also valid.

Reply:
OK, well it's already a dependency from other libraries so I'll pull it in, but note that it doesn't cover all the cases here: not 0.0.0.0 and not localhost (really, no mention of it in their source)

Reply:
I guess it's because it's an IP address manipulation library, not a hostname/DNS library.